testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -3769,6 +3766,120 @@ protected void testColumnName(String columnName, boolean delimited)
         }
     }
 
+    @Test(dataProvider = "testColumnNameDataProvider")
+    public void testAddColumnName(String columnName)
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_ADD_COLUMN));
+
+        if (!requiresDelimiting(columnName)) {
+            testAddColumnName(columnName, false);
+        }
+        testAddColumnName(columnName, true);
+    }
+
+    protected void testAddColumnName(String columnName, boolean delimited)
+    {
+        String nameInSql = toColumnNameInSql(columnName, delimited);
+        String tableName = "tcn_" + nameInSql.toLowerCase(ENGLISH).replaceAll("[^a-z0-9]", "") + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + "(key varchar(50))");
+        assertTableColumnNames(tableName, "key");
+
+        try {
+            assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN " + nameInSql + " varchar(50)");
+        }
+        catch (RuntimeException e) {
+            if (isColumnNameRejected(e, columnName, delimited)) {
+                // It is OK if give column name is not allowed and is clearly rejected by the connector.
+                return;
+            }
+            throw e;
+        }
+
+        assertTableColumnNames(tableName, "key", columnName.toLowerCase(ENGLISH));
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "testColumnNameDataProvider")
+    public void testRenameColumnName(String columnName)
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_RENAME_COLUMN));
+
+        if (!requiresDelimiting(columnName)) {
+            testRenameColumnName(columnName, false);
+        }
+        testRenameColumnName(columnName, true);
+    }
+
+    protected void testRenameColumnName(String columnName, boolean delimited)
+    {
+        String nameInSql = toColumnNameInSql(columnName, delimited);
+        String tableName = "tcn_" + nameInSql.replaceAll("[^a-z0-9]", "") + randomTableSuffix();
+
+        try {
+            assertUpdate("CREATE TABLE " + tableName + "(" + nameInSql + " varchar(50))");
+        }
+        catch (RuntimeException e) {
+            if (isColumnNameRejected(e, columnName, delimited)) {
+                // It is OK if give column name is not allowed and is clearly rejected by the connector.
+                return;
+            }
+            throw e;
+        }
+        assertTableColumnNames(tableName, columnName.toLowerCase(ENGLISH));
+
+        String newNameInSql = toColumnNameInSql(columnName + "_new", delimited);
+
+        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN " + nameInSql + " TO " + newNameInSql);
+        assertTableColumnNames(tableName, columnName.toLowerCase(ENGLISH) + "_new");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "testColumnNameDataProvider")
+    public void testDropColumnName(String columnName)
```

Comment:
Let's merge this with `testAddColumnName`, since the setup isn't trivial (CREATE + exception handling)

Reply:
Do you mean `testRenameColumnName`?  The setup in `testAddColumnName` creates a table without error handling.

Reply:
I meant to merge this with `testAddColumnName`.

that would mean two lines more in that test

```
assertUpdate("ALTER TABLE " + tableName + " DROP COLUMN " + nameInSql);
        assertTableColumnNames(tableName, "value");
```

ws the whole boilerplate of this method.

You can also rename the test method to testAddAndDropColumnName